---
title: "Mutation Saturation at mCpG sites" #date: Nov 16 2021
output:
  html_document: default
  pdf_document: default
---

## INITIALIZE (libraries and functions)
```{r, include = FALSE, echo=FALSE}

options(scipen=999)
library(here)
library(data.table)
library(stringr)
library(ggplot2)
library(ggrepel)
library(splitstackshape)
library(wesanderson)
library(igraph)
library(ggridges)
library(ggsci)
library(gridExtra)
library(gridGraphics)
library(grid)
library(viridis)
library(gtable)
library(ggpubr)
library(cowplot)
library(scales)

```

## MAIN FIGURES
###############################################
### Figure 1 -- 
```{r}

load(here("./out/fig1.Rda"))

p=ggplot(data=plot.df, aes(x=sample.size/1000, y=value, col=pop)) + geom_point(size=2.2) + scale_x_log10(breaks=c(1,5,10,100,280,400,630,780)) + expand_limits(x = 200, y = 1.025) +
 scale_color_manual(values = c("black","seagreen")) +
  theme_bw() + 
geom_text_repel(inherit.aes = T, label=plot.df$name, col= "black", size = 2.5, box.padding = 0.32, segment.size = 0, seed=4,vjust=0.9) +
  ylab("Fraction segregating (synonymous)") + xlab("Haploid sample size (in thousands)") +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(2, 2, 2, 2),
    legend.text = element_text(size=rel(0.8)),
    legend.title = element_blank(),
    axis.text.x = element_text(angle = 60, hjust=1),
    axis.text = element_text(size=8.5),
    axis.title = element_text(size=9))

gt=arrangeGrob(p)
p=as_ggplot(gt)

png(here("./out/fig1.png"), height=4, width=6, units = "in", res=200)
print(p)
dev.off()


pdf(here("./out/Figure 1.pdf"),height=4, width=6, useDingbats=FALSE)
print(p)
dev.off()


```

### Fig 2 -- 
```{r}

load(here("./out/fig2a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper)) + theme_bw()  + geom_hline(yintercept=1)+ 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + ylab("Relative mutation rate") + xlab(NULL) 


load(here("./out/fig2b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper)) + theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + geom_hline(yintercept = 1) + ylab("Fraction of sites segregating") + xlab(NULL) +ylim(0.7,1.01)


load(here("./out/fig2d.Rda"))
p4=ggplot(plot.df) +  geom_pointrange(aes(x=domain.cl, y=est, ymin=lower,ymax=upper, col=csq)) + 
  theme_bw()  +
    theme(
    legend.position = "none",
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_blank(),
    ) + 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  geom_hline(yintercept = 1) +
  scale_color_manual(values=c("#882255","grey60")) +
  ylab("Fraction of sites segregating") + xlab(NULL) +ylim(0.7,1.01)


load(here("./out/fig2c.Rda"))
p3=ggplot(plot.df) +  geom_bar(aes(x=domain.cl, y=count/100, fill=csq), stat="identity", position="dodge", width=0.6, alpha=0.7) +
  scale_fill_manual(values=c("#882255","grey60")) +
  theme_bw() + 
  scale_y_log10() +
  theme(
    legend.position = c(.6, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    legend.key.size = unit(0.8,"line")
    ) +
    theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  ylab("Mutational opportunities (in hundreds)") + xlab(NULL)


#plot
gt=grid.arrange(arrangeGrob(p1 + theme(plot.margin=unit(c(.75,0,0.2,.5),"cm")), p2 + theme(plot.margin=unit(c(.5,0,0.25,0.5),"cm")), ncol=1),
                arrangeGrob(p3+ theme(plot.margin=unit(c(.9,0.5,0.2,.5),"cm")), p4 + theme(plot.margin=unit(c(.5,0.5,0.2,.5),"cm")), ncol=1), ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C","D"), size = 15,
                  x = c(0, 0,0.5,0.5), y = c(1,0.54,1,0.54))

png(here("./out/fig2.png"), height=8, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 2.pdf"),height=8, width=7, useDingbats=FALSE)
print(p)
dev.off()

```

### Figure  3 -- 
```{r}

load(here("./out/fig3a.Rda"))
data.plot = ggplot(data=plot.df, aes(x=sample.size/1000, y=value, col=mcat_label)) + geom_point(size=2) + scale_x_log10(breaks=c(10,100,780)) + expand_limits(x=9,y = 1) +
 scale_color_manual(values = c("black","salmon","goldenrod2")) +
  ylab("Fraction segregating (synonymous)") + xlab("Haploid sample size (in thousands)") +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(0, 0, 1, 1),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5),
    axis.title.x = element_text(vjust=-2),
    axis.title = element_text(size=8.5),
    legend.text = element_text(size=rel(0.65)),
    legend.title = element_blank())


load(here("./out/fig3b.Rda"))
sim.plot = ggplot(data=plot.df, aes(x=sample.size/1000, y=value, col=mut)) + 
  geom_vline(xintercept = 780, lty="dotted", col="grey40") +
  geom_point(size=1.5, pch=1, stroke=1.025) +     
  scale_x_log10(breaks=c(10,100,780,2000,10000))  +
 scale_color_manual(values = c("black","salmon","goldenrod2")) +
  ylab("Fraction segregating (neutral)") + xlab("Haploid sample size (in thousands)") +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(0, 0, 1, 1),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5),
    axis.title = element_text(size=8.5),
    legend.text = element_text(size=rel(0.7)),
    legend.title = element_blank())

lay <- rbind(c(1,12))
gt=arrangeGrob(data.plot + coord_fixed(1.95) + theme(plot.margin = margin(5, 10, 0, 10, "pt")),sim.plot + coord_fixed(1.8) + theme(plot.margin = margin(10, 2, 0, 10, "pt")) , widths=c(1.01,1.48),ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0.4), y = c(1,1))

png(here("./out/fig3.png"), height=3, width=7, units = "in", res=120)
print(p)
dev.off()

pdf(here("./out/Figure 3.pdf"),height=3, width=7, useDingbats=FALSE)
print(p)
dev.off()


```

### Fig 4 -- 
```{r}

load(here("./out/fig4a.Rda"))
p1=ggplot(data = plot.df,
         aes(x = hs, y = as.factor(sample), fill=flag.var.label)) +
  scale_x_log10() +
  scale_fill_manual(values=c("black","#882255","#44AA99","#6699CC")) +
  geom_density_ridges2(alpha=0.65, scale=2) + 
  scale_y_discrete(labels= c("prior","10","100","780","2000","10000")) +
  theme_bw() +  
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7))
    ) +   
  labs(x = "hs", y = "Haploid sample size (in thousands)") +
coord_fixed(1.5) + theme(legend.key.size = unit(0.75,"line"))

load(here("./out/fig4b.Rda"))
p2=ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=2, alpha=0.8) +
  #scale_x_discrete(labels= c("10","100","500","1000","10000")) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.1,1,10,100,1000,10000)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") +
coord_fixed(.7) + theme(legend.key.size = unit(0.75,"line"))


load(here("./out/fig4c.Rda"))
p3=ggplot(data=plot.df) +
  geom_point(aes(x=sample_size_chr/1000, y = mean, col = label),position=position_dodge(width=0), size=plot.df$size) + 
  theme_bw() +
  scale_color_manual(values=c("black","grey60","goldenrod2","indianred","salmon")) +
  ylim(0,1) +
  scale_x_log10(breaks= c(10,100,500,780,1000,2000,10000)) +
  theme_bw() + 
  ylab("Probability segregating")  + xlab("Haploid sample size (in thousands)") +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(2, 2, 2, 2),
    axis.text.x = element_text(angle=60, hjust=1),
    legend.text = element_text(size=rel(0.8)),
    legend.title = element_blank(),
    ) + coord_fixed(1.1)


### PLOT

gt=arrangeGrob(arrangeGrob(p1, p2, nrow=1), p3, nrow=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B", "C"), size = 15,
                  x = c(0, 0.5, 0), y = c(1, 1 ,0.54))

png(here("./out/fig4.png"), height=8, width=7, units = "in", res=120)
print(p)
dev.off()

pdf(here("./out/Figure 4.pdf"),height=8, width=7, useDingbats=FALSE)
print(p)
dev.off()

```

## SUPPLEMENTARY FIGURES
###############################################
### Supp Fig 1 DNM rate by mutation type 
```{r}

load(here("./out/figS1.Rda"))
p=ggplot(plot.df) +  geom_pointrange(aes(x=label, y=rate*1e8, ymin=lower*1e8,ymax=upper*1e8)) + 
  theme_bw()  + coord_fixed(2) +     
  theme(axis.text.x = element_text(angle=60, hjust=1)) +
  scale_y_log10() + 
  geom_text(aes(x=plot.df$label, y=plot.df$rate*1e8), label=round(plot.df$rate*1e8,2), size=3, vjust=1, hjust=-.35)+
  ylab(expression ("De novo mutation rate ("~10^-8~"per site)")) + xlab(NULL)

gt=arrangeGrob(p)
p=as_ggplot(gt)

png(here("./out/figS1.png"), height=4, width=6, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 1-figure supplement 1.pdf"),  height=4, width=6, useDingbats=FALSE)
print(p)
dev.off()


```

### Supp Fig 2 DNM rate by methylation status
```{r}


load(here("./out/figS2a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=cpg.meth.sperm, y=rate*1e8, ymin=lower*1e8,ymax=upper*1e8)) + 
  theme_bw() +
  scale_y_log10() +  ylim(0.1,22)+
    theme(
    axis.text.x = element_text(angle=60, hjust=1),
    ) + 
  ylab(expression ("De novo mutation rate ("~10^-8~"per site)")) + xlab("Average methylation level in testes") 


load(here("./out/figS2b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=cpg.meth.ovary, y=rate*1e8, ymin=lower*1e8,ymax=upper*1e8)) + 
  theme_bw() +
  scale_y_log10() + ylim(0.1,22)+
  theme(
    axis.text.x = element_text(angle=60, hjust=1),
    ) +   
    ylab(expression ("De novo mutation rate ("~10^-8~"per site)")) + xlab("Average methylation level in ovaries") 

# plot
gt=arrangeGrob(p1, p2, ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0.5), y = c(1,1))

png(here("./out/figS2.png"), height=4, width=6, units = "in", res=200)
p
dev.off()

pdf(here("./out/Figure 1-figure supplement 2.pdf"),  height=4, width=6, useDingbats=FALSE)
print(p)
dev.off()

```

### Supp Fig 3 DNM rate by trinuc context
```{r}

load(here("./out/figS3b.Rda"))
p2=ggplot(data=plot.df, aes(x=sample.size/1000, y=value, col=mcat)) + geom_point(size=2) + scale_x_log10(breaks=c(10,100,780)) + expand_limits(x=9, y = 1) +
 scale_color_manual(values = c("black","salmon","goldenrod2","indianred")) +
  ylab("Fraction segregating (synonymous)") + xlab("Haploid sample size (in thousands)") +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(0, 0, 1, 1),
    axis.text.x = element_text(angle=60, hjust=1),
    axis.title.x = element_text(vjust=-2),
    legend.text = element_text(size=rel(0.65)),
    legend.title = element_blank())


load(here("./out/figS3a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=context, y=rate*1e8, ymin=lower*1e8,ymax=upper*1e8)) + 
  theme_bw() +
  scale_y_log10() +  ylim(0.1,22)+
    theme(axis.text.x = element_text(angle=60, hjust=1)) +   
  geom_text(aes(x=plot.df$context, y=plot.df$rate*1e8), label=round(plot.df$rate*1e8,2), size=3, vjust=1, hjust=-.35)+
    ylab(expression ("De novo mutation rate ("~10^-8~"per site)")) + xlab("Trinucleotide context") 

#plot
a=arrangeGrob(p1,p2, nrow=2)

gt=grid.arrange(a)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0), y = c(1.0,0.53))


png(here("./out/figS3.png"), height=7.5, width=5, units = "in", res=200)
p
dev.off()

pdf(here("./out/Figure 1-figure supplement 3.pdf"),  height=7.5, width=5, useDingbats=FALSE)
print(p)
dev.off()

```
### Supp 4 DNMs by exon/non exon 
```{r}

load(here("./out/figS4a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=flag, y=est, ymin=lower,ymax=upper)) + theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + ylab(expression ("De novo mutation rate ("~10^-8~"per site)")) + xlab(NULL)

load(here("./out/figS4b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=repeats, y=est, ymin=lower,ymax=upper, col=label), position = position_dodge(width = 0.1)) + theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + geom_hline(yintercept = 1) + ylab("Relative mutation rate") + xlab("Number of DNMs per site") +  scale_color_manual(values=c("black","salmon","goldenrod2","indianred")) +
  theme_bw() +
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.background = element_rect(colour = "white"),
    legend.margin = margin(2, 2, 2, 2),
    legend.text = element_text(size=rel(0.8)),
    legend.title = element_blank(),
    )

# plot
gt=arrangeGrob(p1+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), p2+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), nrow=1)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0.5), y = c(1,1))

png(here("./out/figS4.png"), height=4, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 1-figure supplement 4.pdf"), height=4, width=7, useDingbats=FALSE)
print(p)
dev.off()

```

### Supp Figure 5 methylation dist
```{r}


load(here("./out/figS5a.Rda"))
p1=ggplot(plot.df) +  geom_bar(aes(x=as.factor(mcat), y=est, fill=csq), stat="identity", position="dodge", width=0.2) +
  scale_fill_manual(values=c("#882255","grey60")) +
  theme_bw() + 
  theme(
    legend.position = c(.05, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(2, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    ) +
  ylab("Fraction of mutational opportunities in methylation bin") + xlab("Methylation level in testes")


load(here("./out/figS5b.Rda"))
p2=ggplot(plot.df) +  geom_bar(aes(x=as.factor(mcat), y=est, fill=csq), stat="identity", position="dodge", width=0.2) +
  scale_fill_manual(values=c("#882255","grey60")) +
  theme_bw() + 
  theme(
    legend.position = c(.05, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(2, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    ) +
  ylab("Fraction of mutational opportunities in methylation bin") + xlab("Methylation level in ovaries")

# plot
gt=arrangeGrob(p1+ theme(plot.margin=unit(c(.8,.5,0.2,.2),"cm")), p2 + theme(plot.margin=unit(c(.8,.2,0.2,.5),"cm")), ncol=2)
p=as_ggplot(gt) +
   draw_plot_label(label = c("A", "B"), size = 15,
                   x = c(0, 0.5), y = c(1,1))

png(here("./out/figS5.png"), height=4.5, width=6, units = "in", res=200)
p
dev.off()

pdf(here("./out/Figure 2-figure supplement 1.pdf"),height=4.5, width=6, useDingbats=FALSE)
print(p)
dev.off()

```


### Supp Figure 6 bvalues mcvicker match for annotations
```{r}

load(here("./out/figS6a.Rda"))
p1=ggplot(plot.df, aes(bval/1000, col=csq)) + stat_ecdf(geom = "point", pad=FALSE) + 
   xlab(expression('B statistic (decreasing background'~ selection%->% ~')')) +
  ylab("Cumulative Proportion") +
  scale_color_manual(values=c("grey60","#882255","black","goldenrod","pink")) +
  theme_bw() + theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)))


load(here("./out/figS6b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper)) + 
  geom_pointrange(data=plot.df, aes(x=csq, y=prev_est, ymin=prev_lower,ymax=prev_upper), pch=4) +
  theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + geom_hline(yintercept = 1) + ylab("Fraction of sites segregating") + xlab(NULL)


load(here("./out/figS6c.Rda"))
p3=ggplot(plot.df, aes(bval/1000, col=domain.cl)) + stat_ecdf(geom = "point", pad=FALSE) +
   xlab(expression('B statistic (decreasing background'~ selection%->% ~')')) + 
  ylab("Cumulative Proportion") +
  scale_color_manual(values=c("grey60","#882255","#44AA99","#6699CC","black","goldenrod2","indianred","salmon")) +
  theme_bw() + theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.key.size = unit(0.75,"line"),
    legend.text = element_text(size=rel(0.7)))


load(here("./out/figS6d.Rda"))
p4=ggplot(plot.df) +  geom_pointrange(aes(x=domain.cl, y=est, ymin=lower,ymax=upper, col=csq)) + 
  geom_point(aes(x=domain.cl, y=prev_est, col=csq), pch=4) + 
  theme_bw()  +
    theme(
    legend.position = c(0.01,0.35),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7))
    ) + 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  geom_hline(yintercept = 1) +
  scale_color_manual(values=c("#882255","grey60")) +
  ylab("Fraction of sites segregating") + xlab(NULL)


gt=grid.arrange(arrangeGrob(p1 + theme(plot.margin=unit(c(.75,0,0.2,.5),"cm")), p2, ncol=1),
                arrangeGrob(p3+ theme(plot.margin=unit(c(.9,0.5,0.2,.5),"cm")), p4 + theme(plot.margin=unit(c(.5,0.5,0.2,.5),"cm")), ncol=1), ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C","D"), size = 15,
                  x = c(0, 0,0.5,0.5), y = c(1,0.54,1,0.54))

png(here("./out/figS6.png"), height=7, width=8, units = "in", res=200)
p
dev.off()


pdf(here("./out/Figure 2-figure supplement 2.pdf"),height=7, width=8, useDingbats=FALSE)
print(p)
dev.off()


```


### Supp Figure 7 LOF location in protein
```{r} 


load(here("./out/figS7a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=flag, y=est, ymin=lower,ymax=upper, col=csq), position = position_dodge(0.1))+ theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
    scale_color_manual(values = c("black","indianred","goldenrod2")) +
  geom_hline(yintercept = 1) + ylab("Relative mutation rate") + xlab(NULL) +
      theme(
    legend.position = c(0.6,0.99),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(),
    )

load(here("./out/figS7b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=flag, y=est, ymin=lower,ymax=upper, col=csq)) + 
  theme_bw()  +
    theme(
    legend.position = "none",
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_blank(),
    ) +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  geom_hline(yintercept = 1) +
  scale_color_manual(values = c("black","indianred","goldenrod2")) +
  ylab("Fraction of sites segregating") + xlab(NULL)


## plot
gt=arrangeGrob(p1+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), p2+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), nrow=1)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0.5), y = c(1,1))

png(here("./out/figS7.png"), height=4, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 2-figure supplement 3.pdf"),height=4, width=7, useDingbats=FALSE)
print(p)
dev.off()

```


### Supp Figure. 8 worst csq version of missense/domain 
```{r}


load(here("./out/figS8a.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper)) + theme_bw()  + geom_hline(yintercept=1)+ 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + ylab("Relative mutation rate") + xlab(NULL) 

load(here("./out/figS8b.Rda"))
p2=ggplot(plot.df) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper)) + theme_bw()  +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + geom_hline(yintercept = 1) + ylab("Fraction of sites segregating") + xlab(NULL)

load(here("./out/figS8d.Rda"))
p4=ggplot(plot.df) +  geom_pointrange(aes(x=domain_all.cl, y=est, ymin=lower,ymax=upper, col=csq)) + 
  theme_bw()  +
    theme(
    legend.position = "none",
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_blank(),
    ) + 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  geom_hline(yintercept = 1) +
  scale_color_manual(values=c("#882255","grey60")) +
  ylab("Fraction of sites segregating") + xlab(NULL)

load(here("./out/figS8c.Rda"))
p3=ggplot(plot.df) +  geom_bar(aes(x=domain_all.cl, y=count/100, fill=csq), stat="identity", position="dodge", width=0.6, alpha=0.7) +
  scale_fill_manual(values=c("#882255","grey60")) +
  theme_bw() + 
  scale_y_log10() +
  theme(
    legend.position = c(.6, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    legend.key.size = unit(0.8,"line")
    ) +
    theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
  ylab("Mutational opportunities (in hundreds)") + xlab(NULL)

#plot

gt=grid.arrange(arrangeGrob(p1 + theme(plot.margin=unit(c(.75,0,0.2,.5),"cm")), p2, ncol=1),
                arrangeGrob(p3+ theme(plot.margin=unit(c(.9,0.5,0.2,.5),"cm")), p4 + theme(plot.margin=unit(c(.5,0.5,0.2,.5),"cm")), ncol=1), ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C","D"), size = 15,
                  x = c(0, 0,0.5,0.5), y = c(1,0.54,1,0.54))

png(here("./out/figS8.png"), height=8, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 2-figure supplement 4.pdf"),height=8, width=7, useDingbats=FALSE)
print(p)
dev.off()



```


### Supp Figure 9 DNMs by CADD
```{r}


load(here("./out/figS9ab.Rda"))
p1=ggplot(plot.df) +  geom_pointrange(aes(x=as.factor(round(cadd_edge,1)), y=rate/total, ymin=lower/total,ymax=upper/total)) + 
  theme_bw() + ylim(c(0.5,1.5)) + geom_hline(yintercept=1) +
  ylab("Relative de novo mutation rate") + xlab("Upper limit of CADD decile")

p2=ggplot(plot.df)  + 
  geom_point(aes(x=as.factor(round(cadd_edge,1)), y=frac_segregating), col="indianred",size=3) +
  theme_bw() + ylim(c(0.7,1)) +
  ylab("Fraction of sites segregating") + xlab("Upper limit of CADD decile")


load(here("./out/figS9cd.Rda"))
p3=ggplot(plot.df) +  
  geom_pointrange(aes(x=as.factor(round(cadd_edge,1)), y=rate/total, ymin=lower/total,ymax=upper/total)) + 
  theme_bw() + ylim(c(0.5,1.5)) + geom_hline(yintercept=1) +
  ylab("Relative de novo mutation rate") + xlab("Upper limit of CADD decile")

p4=ggplot(plot.df)  + 
  geom_point(aes(x=as.factor(round(cadd_edge,1)), y=frac_segregating), col="indianred",size=3) +
  theme_bw() + ylim(c(0.7,1)) +
  ylab("Fraction of sites segregating") + xlab("Upper limit of CADD decile")

# plot
gt=grid.arrange(arrangeGrob(p2, p1, nrow=2),arrangeGrob(p4, p3, nrow=2), ncol=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C","D"), size = 15,
                  x = c(0, 0, 0.5, 0.5), y = c(1,0.54,1,0.54))

png(here("./out/figS9.png"), height=7, width=9, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 2-figure supplement 5.pdf"),height=7, width=9, useDingbats=FALSE)
print(p)
dev.off()

```

### Supp Figure 10a, b,c
```{r}

load(here("./out/figS10a.Rda"))
p1=  ggplot(data = plot.df, aes(shape=mu_label)) +
  geom_point( aes(x = sample.size/1000, y = mu*(mean_tree_length), ymin=mu*(mean_tree_length-2*sd_tree_length), ymax=mu*(mean_tree_length+2*sd_tree_length)), fill="white", stroke=1.025, col="black") +     
  scale_y_log10() + geom_hline(yintercept=1, lty="dashed") +
  theme_bw() +  
  scale_x_log10(breaks=c(10,100,780,2000,10000))  +
  #scale_color_manual(values = c("black","salmon","goldenrod2")) +
  scale_shape_manual(values = c(19, 21, 23)) +
  guides(color = "none") + theme(legend.key.size = unit(10, 'pt')) + 
  theme(
    legend.position = c(.01, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7))
    ) + coord_fixed(0.3) +
  labs(x = "Haploid sample size (in thousands)", y = "Expected neutral mutations per site")


load(here("./out/figS10c.Rda"))
p3=ggplot(data = plot.df,
  aes(x = sample.size/1000, y = mean_tree_length/1e6, col=label)) +
  geom_point(size=2.5)+
  scale_color_manual(values=c("indianred","goldenrod2","black")) +
#geom_pointrange(aes(ymin=(mean_tree_length-2*sd_tree_length)/1e6, ymax=(mean_tree_length+2*sd_tree_length)/1e6)) +
  theme_bw() + theme(legend.key.size = unit(10, 'pt')) +  
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.6))
    )  + scale_x_log10(breaks=c(10, 100, 250, 500, 1000, 2500)) + scale_y_log10() + coord_fixed(1.2)+
  labs(x = "Haploid sample size (in thousands)", y = "Mean tree length (millions of generations)")


load(here("./out/figS10b.Rda"))
p2=ggplot(data = plot.df,
  aes(x = sample.size/1000, y = mean_tree_length/1e6, col=as.factor(label))) +
  geom_point(size=2.5) +
  scale_x_log10(breaks=c(10, 100, 250, 500, 1000, 2500)) + scale_y_log10() +
  scale_color_manual(values=c("indianred","goldenrod2","black","grey60")) +
#geom_pointrange(aes(ymin=(mean_tree_length-2*sd_tree_length)/1e6, ymax=(mean_tree_length+2*sd_tree_length)/1e6)) +
  theme_bw() + theme(legend.key.size = unit(10, 'pt')) +  
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.6))
    ) + coord_fixed(1.1)  +
  labs(x = "Haploid sample size (in thousands)", y = "Mean tree length (millions of generations)")


#plot
gt=arrangeGrob(p1+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), arrangeGrob(p2 + theme(plot.margin = margin(20, 5, 0, 5, "pt")), p3+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), nrow=1), nrow=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B", "C"), size = 15,
                  x = c(0, 0, 0.5), y = c(1,0.5 ,0.5))

png(here("./out/figS10.png"), height=7, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 3 - figure supplement 1.pdf"), height=7, width=7, useDingbats=FALSE)
print(p)
dev.off()


```
 
### Supp Figure 11 
```{r}

load(here("./out/figS11a.Rda"))
p1=ggplot(plot.df) +  geom_point(aes(x=csq, y=count), size=10, shape=95, col="black") + theme_bw() +
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + ylab("Number of sites per bin (k)") + xlab(NULL) 

load(here("./out/figS11bc.Rda"))
p2=ggplot(plot.df[csq=="synonymous"]) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper, shape=mcat, col=mcat, fill=mcat)) + 
  theme_bw()  + 
  ylim(0.7,1) +
   theme(legend.position="none",axis.text.x = element_text(angle = 60, hjust=1)) + 
  ylab("Fraction of bins segregating") + xlab(NULL) + 
  scale_shape_manual(values=c(21,4)) +
  scale_fill_manual(values=c("black","red")) +
  scale_color_manual(values=c("black","red"))

p3=ggplot(plot.df[!csq=="synonymous"]) +  geom_pointrange(aes(x=csq, y=est, ymin=lower,ymax=upper,shape=mcat, col=mcat, fill=mcat)) + theme_bw()  + 
   theme(legend.position="none",axis.text.x = element_text(angle = 60, hjust=1)) + ylab("Fraction of bins segregating") + xlab(NULL) +   ylim(0.7,1) +  scale_shape_manual(values=c(21,4)) +
  scale_fill_manual(values=c("black","red")) +
  scale_color_manual(values=c("black","red"))

gt=arrangeGrob(p1+ theme(plot.margin = margin(5, 5, 0, 5, "pt")), arrangeGrob(p2 + theme(plot.margin = margin(5, 5, 0, 5, "pt")), p3 + theme(plot.margin = margin(5, 5, 0, 5, "pt")), nrow=1, widths=c(1,1.9)), nrow=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C"), size = 15,
                  x = c(0, 0, 0.33), y = c(1,0.53, 0.53))

png(here("./out/figS11.png"), height=7, width=6, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 3 - figure supplement 2.pdf"), height=7, width=6, useDingbats=FALSE)
print(p)
dev.off()


```
### Supp Figure 12 other priors 
#### Figure odds plots -- 1e-7 mut rate gamma prior
```{r}

load(here("./out/figS12a1.Rda"))
prior.plot=ggplot(data=plot.df, aes(x=log10(hs))) + 
  geom_histogram(aes(y = ..density..), color = "grey30", fill = "grey60") +
  theme_bw() + theme(axis.text=element_text(size=10))


load(here("./out/figS12a2.Rda"))
post.plot=ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=3, alpha=0.8) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.1,1,10,100,1000,10000)) + expand_limits(y = c(0.1,10000)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1,size=10),
    axis.text=element_text(size=10)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") + theme(legend.key.size = unit(0.75,"line"))

a=arrangeGrob(prior.plot+ theme(plot.margin=unit(c(0.5,0.2,0.2,0.2),"cm"))
,post.plot + theme(plot.margin=unit(c(0.5,0.2,0.2,0.2),"cm")),nrow=1)

```

#### Figure odds plots -- 1e-7 mut rate lognormal prior
```{r}
load(here("./out/figS12b1.Rda"))
prior.plot=ggplot(data=plot.df, aes(x=log10(hs))) + 
  geom_histogram(aes(y = ..density..), color = "grey30", fill = "grey60") +
  theme_bw() + theme(axis.text=element_text(size=10))


load(here("./out/figS12b2.Rda"))
post.plot=ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=3, alpha=0.8) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.1,1,10,100,1000,10000)) + expand_limits(y = c(0.1,10000)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1,size=10),
        axis.text.y = element_text(size=10)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") + theme(legend.key.size = unit(0.75,"line")) 

b=arrangeGrob(prior.plot, post.plot, nrow=1)

```

#### Figure odds plots -- 1e-7 mut rate beta prior
```{r}

load(here("./out/figS12c1.Rda"))
prior.plot=ggplot(data=plot.df, aes(x=log10(hs))) + 
  geom_histogram(aes(y = ..density..), color = "grey30", fill = "grey60") +
  theme_bw() + theme(axis.text=element_text(size=10))


load(here("./out/figS12c2.Rda"))
post.plot = ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=3, alpha=0.8) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.01,0.1,1,10,100,1000,10000)) + expand_limits(y = c(0.1,10)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1, size=10),
    axis.text.y = element_text(size=10)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") + theme(legend.key.size = unit(0.75,"line"))


c=arrangeGrob(prior.plot,post.plot, nrow=1)

gt=grid.arrange(a,b,c, nrow=3)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B","C"), size = 15,
                  x = c(0, 0, 0), y = c(1,0.7,0.35))

png(here("./out/figS12.png"), height=8, width=8, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 4-figure supplement 1.pdf"), height=8, width=8, useDingbats=FALSE)
print(p)
dev.off()



```
### Supp Figure 13 clinvar and DDD
```{r}

load(file=here("./out/figS13.Rda"))
p1=ggplot(plot.df) +  geom_point(aes(x=label, y=bayes_odds, col=seg), size=2, alpha=0.8) + expand_limits(y=c(0.1,60)) +
    scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
    theme_bw()  + scale_y_log10() +
    theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    legend.key.size = unit(0.75,"line")
    ) + 
   theme(axis.text.x = element_text(angle = 60, hjust=1)) + 
    ylab("Odds of variant being pathogenic") + xlab(NULL) + 
    geom_hline(yintercept = 1, col="black", lty="dashed") 


gt=arrangeGrob(p1+ theme(plot.margin = margin(5, 5, 5, 5, "pt")))
p=as_ggplot(gt)

png(here("./out/figS13.png"), height=4, width=2.5, units = "in", res=120)
print(p)
dev.off()

pdf(here("./out/Figure 4-figure supplement 2.pdf"),  height=4, width=2.5, useDingbats=FALSE)
print(p)
dev.off()

```
### Supp Figure 14 other mut rate rate abc plots 
#### 1e-9 mutation rate uniform prior
```{r}

load(here("./out/figS14a1.Rda"))
p1=ggplot(data = plot.df,
         aes(x = hs, y = as.factor(sample), fill=flag.var.label)) +
  scale_x_log10() + expand_limits(y = c(0,10)) +
  scale_fill_manual(values=c("black","#882255","#44AA99","#6699CC")) +
  geom_density_ridges(alpha = .65, scale=0.8, rel_min_height = 0.01) +
  scale_y_discrete(labels= c("prior","10","100","780","2000","10000")) +
  theme_bw() +  
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7))
    ) +   
  labs(x = "hs", y = "Haploid sample size (in thousands)") +
coord_fixed(1.5) + theme(legend.key.size = unit(0.75,"line"))


load(here("./out/figS14a2.Rda"))
p2=ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  #scale_fill_manual(values=wes_palette(n=3, name="Royal1")) +
  #scale_color_manual(values=pal_npg("nrc", alpha = 0.7)(4)[c(1,3,4)]) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=2.5, alpha=0.8) +
  #scale_x_discrete(labels= c("10","100","500","1000","10000")) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.1,1,10)) + expand_limits(y = c(0.1,10)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") + theme(legend.key.size = unit(0.75,"line"))

```
#### 1e-8 mutation rate uniform prior
```{r}

load(here("./out/figS14b1.Rda"))
p3=ggplot(data = plot.df,
         aes(x = hs, y = as.factor(sample), fill=flag.var.label)) +
  scale_x_log10() + expand_limits(y = c(0,10)) +
  scale_fill_manual(values=c("black","#882255","#44AA99","#6699CC")) +
  geom_density_ridges(alpha = .65, scale=0.8, rel_min_height = 0.01) +
  scale_y_discrete(labels= c("prior","10","100","780","2000","10000")) +
  theme_bw() +  
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7))
    ) +   
  labs(x = "hs", y = "Haploid sample size (in thousands)") +
coord_fixed(1.5) + theme(legend.key.size = unit(0.75,"line"))

load(here("./out/figS14b2.Rda"))
p4=ggplot(data = plot.df,
         aes(y = (over/under)/(prior_over/prior_under), x = sample, col=as.factor(flag.var.label))) +
  #scale_fill_manual(values=wes_palette(n=3, name="Royal1")) +
  #scale_color_manual(values=pal_npg("nrc", alpha = 0.7)(4)[c(1,3,4)]) +
  scale_color_manual(values=c("#882255","#44AA99","#6699CC")) +
  geom_point(size=2.5, alpha=0.8) +
  #scale_x_discrete(labels= c("10","100","500","1000","10000")) +
  theme_bw() +  
  geom_hline(yintercept = 1, col="black", lty="dashed") +
  scale_y_log10(breaks= c(0.1,1,10)) + expand_limits(y = c(0.1,10)) +
  scale_x_log10(breaks= c(10,100,780,2000,10000)) +
  theme(
    legend.position = c(.02, .99),
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    axis.text.x = element_text(angle=60, hjust=1, size=8.5)
    ) +   
  labs(y = "Bayes odds s > 1e-3", x = "Haploid sample size (in thousands)") + theme(legend.key.size = unit(0.75,"line"))

```

#### plot
```{r}

a=arrangeGrob(p1 + theme(plot.margin = margin(10, 10, 10, 20, "pt")), p2+ theme(plot.margin = margin(10, 10, 10, 10, "pt")), nrow=1)
b=arrangeGrob(p3 + theme(plot.margin = margin(10, 10, 10, 20, "pt")),p4+ theme(plot.margin = margin(10, 10, 10, 10, "pt")), nrow=1)

gt=grid.arrange(a,b, nrow=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B"), size = 15,
                  x = c(0, 0), y = c(1,0.53))


png(here("./out/figS14.png"), height=7, width=6.5, units = "in", res=200)
p
dev.off()

pdf(here("./out/Figure 4-figure supplement 3.pdf"),  height=7, width=6.5, useDingbats=FALSE)
print(p)
dev.off()

```

### Supp Figure 15 -- Cpg Consequences vs other mutation types
```{r}


load(here("./out/figS15c.Rda"))
plot3=ggplot(plot.df, aes(CADD_phred, col=mcat)) + stat_ecdf(geom = "point", pad=FALSE) + 
  xlab("CADD score") + ylab("Cumulative Proportion") +
  scale_color_manual(values=c("grey60","#882255")) +
  theme_bw() + theme(
    legend.position = "none",
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)))


load(here("./out/figS15a.Rda"))
plot1=ggplot(plot.df) +  geom_bar(aes(x=csq, y=est, fill=label), stat="identity", position="dodge", width=0.7) +
  scale_fill_manual(values=c("#882255","grey60")) +
  theme_bw() + 
  theme(
    legend.position = c(.90, .99),
    legend.justification = c("right", "top"),
    legend.box.just = c("right", "top"),
    legend.margin = margin(2, 2, 2, 2),
    legend.title = element_blank(),
    legend.text = element_text(size=rel(0.7)),
    ) +
  ylab("Fraction of mutational opportunities in annotation") + xlab(NULL)


load(here("./out/figS15b.Rda"))
plot2=ggplot(plot.df, aes(bval/1000, col=mcat)) + stat_ecdf(geom = "point", pad=FALSE) + 
   xlab("B statistic") +
  ylab("Cumulative Proportion") +
  scale_color_manual(values=c("grey60","#882255")) +
  theme_bw() + theme(
    legend.position = "none",
    legend.justification = c("left", "top"),
    legend.box.just = c("left", "top"),
    legend.margin = margin(1, 2, 2, 2),
    legend.title = element_blank(),
    legend.key.size = unit(0.9,"line"),
    legend.text = element_text(size=rel(0.9)))

gt=arrangeGrob(plot1+ theme(plot.margin = margin(20, 20, 0, 20, "pt")), arrangeGrob(plot2 + theme(plot.margin = margin(20, 5, 0, 5, "pt")), plot3+ theme(plot.margin = margin(20, 5, 0, 5, "pt")), nrow=1), nrow=2)
p=as_ggplot(gt) +
  draw_plot_label(label = c("A", "B", "C"), size = 15,
                  x = c(0, 0, 0.5), y = c(1,0.5 ,0.5))

png(here("./out/figS15.png"), height=7, width=7, units = "in", res=200)
print(p)
dev.off()

pdf(here("./out/Figure 4-figure supplement 4.pdf"),  height=7, width=7, useDingbats=FALSE)
print(p)
dev.off()

```




